lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,234 @@
+        encode_string(output, key)?;
+        output.write_char('=')?;
+        encode_value(output, value)
+    }
+
+    pub fn encode(input: &BTreeMap<String, Value>, fields: &[String]) -> Result<String, String> {
+        let mut output = String::new();
+        let mut seen_fields = HashSet::new();
+
+        for (idx, field) in fields.iter().enumerate() {
+            if let Some(val) = input.get(field) {
+                if idx > 0 {
+                    output.write_char(' ').map_err(|_| "write error")?;
+                }
+                encode_field(&mut output, field, val).map_err(|_| "write error")?;
+                seen_fields.insert(field);
+            }
+        }
+
+        for (idx, (key, value)) in input.iter().enumerate() {
+            if seen_fields.contains(key) {
+                continue;
+            }
+
+            if idx > 0 || !seen_fields.is_empty() {
+                output.write_char(' ').map_err(|_| "write error")?;
+            }
+            encode_field(&mut output, key, value).map_err(|_| "write error")?;
+        }
+
+        Ok(output)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeLogfmt;
+
+impl Function for EncodeLogfmt {
+    fn identifier(&self) -> &'static str {
+        "encode_logfmt"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::OBJECT,
+                required: true,
+            },
+            Parameter {
+                keyword: "fields_ordering",
+                kind: kind::ARRAY,
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        let fields = arguments.optional("fields_ordering");
+
+        Ok(Box::new(EncodeLogfmtFn { value, fields }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "encode object",
+                source: r#"encode_logfmt!({"lvl": "info", "msg": "This is a message", "log_id": 12345})"#,
+                result: Ok(r#"s'log_id=12345 lvl=info msg="This is a message"'"#),
+            },
+            Example {
+                title: "encode object with fields ordering",
+                source: r#"encode_logfmt!({"msg": "This is a message", "lvl": "info", "log_id": 12345}, ["lvl", "msg"])"#,
+                result: Ok(r#"s'lvl=info msg="This is a message" log_id=12345'"#),
+            },
+        ]
+    }
+}
+
+#[derive(Clone, Debug)]
+struct EncodeLogfmtFn {
+    value: Box<dyn Expression>,
+    fields: Option<Box<dyn Expression>>,
+}
+
+fn resolve_fields(fields: &Value) -> Vec<String> {
+    match fields {
+        Value::Array(arr) => arr
+            .iter()
+            .filter_map(|v| match v {
+                Value::Bytes(bytes) => Some(String::from_utf8_lossy(&bytes)),
+                _ => None,
```

Comment:
I think it'd be better to return an explicit error here if the array contains a value that is not a string.